runtime/gc_base/GCExtensions.cpp

```diff
@@ -231,35 +231,34 @@ void
 MM_GCExtensions::computeDefaultMaxHeap(MM_EnvironmentBase *env)
 {
 	OMRPORT_ACCESS_FROM_OMRPORT(env->getPortLibrary());
-	uint64_t cgroupMemory = 0;
 
 	MM_GCExtensionsBase::computeDefaultMaxHeap(env);
 
 	if (OMR_CGROUP_SUBSYSTEM_MEMORY == omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_MEMORY)) {
-/* A very rough estimate of the minimum amount of memory required by the JVM excluding
- * the heap. The size is a conservative estimate, attempting to leave room for the JVM's
- * internal requirements given that one compilation thread can use up to 256M.
- */
-#define OPENJ9_IN_CGROUP_NATIVE_FOOTPRINT_EXCLUDING_HEAP ((U_64)512 * 1024 * 1024)
-
+		uint64_t cgroupMemory = 0;
 		int32_t rc = omrsysinfo_cgroup_get_memlimit(&cgroupMemory);
```

Comment:
Why are we calling this API since we are not using the value?

Reply:
We actually need to determine if cgroup memory limit is set or not. We currently don't have any API for that but we can achieve that using `omrsysinfo_cgroup_get_memlimit`. We are only interested in its return value.

Reply:
Hmmmmm... since we want to know this shouldn't we add an API for that?  The API you are calling seems like overkill and not very useful.  Maybe we should delete this API and create one that just gives you the information you are looking for.

Reply:
well, actually an API to determine if cgroup memory limit is set or not would end up doing the same set of steps that `omrsysinfo_cgroup_get_memlimit` is doing. Only difference would be it wouldn't return the memory limit, but just return true or false.

Reply:
Probably I shouldn't have added `omrsysinfo_cgroup_get_memlimit()` as an API, but some internal function (since it is used by `sysinfo_get_physical_memory`) and exposed it only when required.
Currently we don't have any other use for this API, but it could be used for https://github.com/eclipse/openj9/issues/1932.
If you feel strongly about removing this API, then I can do that, convert it into an internal function and a new API for determining if cgroup memory limit is set or not.

Reply:
I think it would be clearer if there was an API to query if a cgroup memory limit is set.

Reply:
Created https://github.com/eclipse/omr/pull/2430 for adding the API to query if a cgroup memory limit is set.